parachains/runtimes/collectives/collectives-polkadot/src/ambassador/mod.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The Ambassador Program.
+//!
+//! The module defines the following on-chain functionality of the Ambassador Program:
+//! - managed set of the program members, where every member has a [rank](ranks) (via [pallet_ranked_collective]).
+//! - referendum functionality for the program members to propose, vote and execute passed proposals on behalf
+//! of the members of a certain [rank](Origin) (via [pallet_referenda]).
+//! - managed content (charter, announcements) (via [pallet_collective_content]).
+
+pub mod origins;
+mod tracks;
+
+use super::*;
+use frame_support::traits::TryMapSuccess;
+pub use origins::pallet_origins as pallet_ambassador_origins;
+use origins::pallet_origins::{
+	EnsureAmbassador, EnsureRankedAmbassador, EnsureSeniorAmbassador, Origin,
+};
+use sp_arithmetic::traits::CheckedSub;
+use sp_runtime::{
+	morph_types,
+	traits::{ConstU16, TypedGet},
+};
+
+/// The Ambassador Program members ranks.
+pub mod ranks {
+	use pallet_ranked_collective::Rank;
+
+	#[allow(dead_code)]
+	pub const CANDIDATE: Rank = 0;
+	pub const AMBASSADOR: Rank = 1;
+	pub const SENIOR_AMBASSADOR: Rank = 2;
+	pub const HEAD_AMBASSADOR: Rank = 3;
+}
+
+pub type AmbassadorContentInstance = pallet_collective_content::Instance1;
+
+impl pallet_collective_content::Config<AmbassadorContentInstance> for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type CharterOrigin = EnsureAmbassador;
+	type AnnouncementOrigin = pallet_ranked_collective::EnsureMember<
+		Runtime,
+		AmbassadorCollectiveInstance,
+		{ ranks::AMBASSADOR },
```

Comment:
Does this mean that any `Ambassador` can make announcements? That might be a little too open, because it'd let new member make announcements that appear to represent the program. I'd say probably "Single HeadAmbassador OR Referendum of Ambassador Origin".